autoendpoint/src/routers/fcm/client.rs

```diff
@@ -153,15 +153,20 @@ impl FcmClient {
                 (StatusCode::NOT_FOUND, _) => RouterError::NotFound,
                 (_, Some(error)) => {
                     info!("🌉Bridge Error: {:?}, {:?}", error.message, &self.endpoint);
-                    RouterError::Upstream {
-                        status: error.status,
+                    FcmError::Upstream {
+                        error_code: error.status, // Note: this is the FCM error status enum value
                         message: error.message,
                     }
+                    .into()
                 }
-                (status, None) => RouterError::Upstream {
-                    status: status.to_string(),
-                    message: "Unknown reason".to_string(),
-                },
+                // In this case, we've gotten an error, but FCM hasn't returned a body.
+                // (This may happen in the case where FCM terminates the connection abruptly
+                // or a similar event.) Treat that as an INTERNAL error.
+                (_, None) => FcmError::Upstream {
+                    error_code: "INTERNAL".to_string(),
```

Comment:
nit: how about distinguishing between their `INTERNAL` type w/ the response's status code (e.g. `reason: UNKNOWN_502`)
```suggestion
                    error_code: format!("UNKNOWN_{}", status.as_str()),
```

Reply:
I'm fine using "UNKNOWN" but I don't think adding the status code will get us much more. 
I'm going to add a `warn!()` logger message to report the error, status and the raw response, since that will probably be more useful when we see a bunch of these, since they should be (hopefully) very rare.